sway-core/src/convert_parse_tree.rs

```diff
@@ -514,11 +546,32 @@ fn item_enum_to_enum_declaration(
         .enumerate()
         .map(|(tag, type_field)| type_field_to_enum_variant(ec, type_field, tag))
         .collect::<Result<Vec<_>, _>>()?;
+
     if variants.iter().any(|variant| {
        matches!(&variant.r#type, TypeInfo::Custom { name, ..} if name == &item_enum.name)
     }) {
-        return Err(ec.error(ConvertParseTreeError::RecursiveType { span }));
+        errors.push(ConvertParseTreeError::RecursiveType { span: span.clone() });
+    }
+
+    // Make sure each variant is declared once
+    let mut map = BTreeMap::new();
+    variants.iter().rev().for_each(|f| {
+        map.entry(&f.name).or_insert(vec![]).push(f.clone());
+    });
+
+    map.iter()
+        .filter(|(_, occurences)| occurences.len() > 1)
+        .for_each(|(name, _)| {
+            errors.push(ConvertParseTreeError::DuplicateEnumVariant {
+                name: (*name).clone(),
+                span: name.span().clone(),
+            });
+        });
```

Comment:
Yeah I'll try that out. The `rev` is because I want the last occurrence to show the error but I guess that's not too important. Ideally I'd print the error for each illegal occurrence and use a multi span that also points to the first occurrence.

Reply:
Love it 😍

Reply:
Done. Much better and in fact I now get error messages for _each_ duplicate of the same name.